contracts/stellar-interchain-token-service/src/migrate.rs

```diff
@@ -46,56 +43,60 @@ impl CustomMigratableInterface for InterchainTokenService {
-            token_ids,
             new_token_manager_wasm_hash,
             new_interchain_token_wasm_hash,
         } = migration_data;
 
         storage::set_token_manager_wasm_hash(env, &new_token_manager_wasm_hash);
         storage::set_interchain_token_wasm_hash(env, &new_interchain_token_wasm_hash);
 
-        let current_epoch = current_epoch(env);
-        let upgrader_client = UpgraderClient::new(env, &upgrader_client);
-
-        for token_id in token_ids.into_iter() {
-            let TokenIdConfigValue {
-                token_address: interchain_token,
-                token_manager,
-                token_manager_type,
-            } = storage::try_token_id_config(env, token_id.clone())
-                .ok_or(ContractError::InvalidTokenId)?;
-
-            upgrader_client.upgrade(
-                &token_manager,
-                &new_version,
-                &new_token_manager_wasm_hash,
-                &stellar_axelar_std::vec![env, ().into()],
-            );
-
-            if token_manager_type != TokenManagerType::LockUnlock {
-                upgrader_client.upgrade(
-                    &interchain_token,
-                    &new_version,
-                    &new_interchain_token_wasm_hash,
-                    &stellar_axelar_std::vec![env, ().into()],
-                );
-            }
-
-            let flow_key = legacy_storage::FlowKey {
-                token_id: token_id.clone(),
-                epoch: current_epoch,
-            };
-
-            if let Some(flow_out) = legacy_storage::try_flow_out(env, flow_key.clone()) {
-                storage::set_flow_out(env, token_id.clone(), current_epoch, &flow_out);
-            }
-            if let Some(flow_in) = legacy_storage::try_flow_in(env, flow_key.clone()) {
-                storage::set_flow_in(env, token_id.clone(), current_epoch, &flow_in);
-            }
-        }
-
         Ok(())
     }
 }
+
+pub fn migrate_token(
+    env: &Env,
+    token_id: BytesN<32>,
+    upgrader_client: Address,
+    new_version: String,
+) -> Result<(), ContractError> {
+    let current_epoch = current_epoch(env);
+    let upgrader_client = UpgraderClient::new(env, &upgrader_client);
+
+    let TokenIdConfigValue {
+        token_address: interchain_token,
+        token_manager,
+        token_manager_type,
+    } = storage::try_token_id_config(env, token_id.clone()).ok_or(ContractError::InvalidTokenId)?;
+
+    upgrader_client.upgrade(
+        &token_manager,
+        &new_version,
+        &storage::token_manager_wasm_hash(env),
+        &vec![env, ().into()],
+    );
+
+    /* Only tokens deployed via ITS may be upgraded. */
+    if token_manager_type == TokenManagerType::NativeInterchainToken {
+        upgrader_client.upgrade(
+            &interchain_token,
+            &new_version,
+            &storage::interchain_token_wasm_hash(env),
+            &vec![env, ().into()],
+        );
+    }
+
+    let flow_key = legacy_storage::FlowKey {
```

Comment:
```suggestion
    let current_epoch = current_epoch(env);
    let flow_key = legacy_storage::FlowKey {
```
move the definition to where it's needed